recipes/service.rb

```diff
@@ -6,12 +6,14 @@
 template 'marathon-init' do
   path   '/etc/init/marathon.conf'
   source 'upstart.erb'
-  variables(wrapper: ::File.join(node['marathon']['home'], 'wrapper'))
+  variables(wrapper: ::File.join(node['marathon']['home'], 'wrapper'),
+            user:    node['marathon']['user'])
 end
 
 service 'marathon' do
   provider   Chef::Provider::Service::Upstart
   supports   status: true, restart: true
-  subscribes :restart, 'template[marathon-init]'
+  subscribes :stop, 'template[marathon-init]'
+  subscribes :start, 'template[marathon-init]'
```

Comment:
Let's consider the following example:
* Somewhere else before creating this resource, a `subscribes :start` (delayed) is done
* Then here we subscribe to `stop` then `start` (still delayed), but since the `start` has already been subscribed, this `subscribes :start` has no effect.
As a result we end up having the `start` action being executed before the `stop` action, and the Marathon service is stopped at the end of the chef run.
In order to avoid this, we may force the `stop` and `start` to be done immediately ..